wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsworkunits/WsWorkunitsClientSoapProxyWrapper.java

```diff
@@ -0,0 +1,105 @@
+import org.hpccsystems.ws.client.gen.wsworkunits.v1_73.ArrayOfEspException;
+import org.hpccsystems.ws.client.gen.wsworkunits.v1_73.EspException;
+import org.hpccsystems.ws.client.gen.wsworkunits.v1_73.WsWorkunitsServiceSoap;
+import org.hpccsystems.ws.client.gen.wsworkunits.v1_73.WsWorkunitsServiceSoapProxy;
+import org.hpccsystems.ws.client.platform.Version;
+import org.hpccsystems.ws.client.platform.WorkunitInfo;
+import org.hpccsystems.ws.client.utils.Connection;
+
+public class WsWorkunitsClientSoapProxyWrapper
+{
+    
+    private org.hpccsystems.ws.client.gen.wsworkunits.v1_56.WsWorkunitsServiceSoap version5WorkunitsServiceSoap = null;
+    private org.hpccsystems.ws.client.gen.wsworkunits.v1_69.WsWorkunitsServiceSoap version6WorkunitsServiceSoap = null;
+    private org.hpccsystems.ws.client.gen.wsworkunits.v1_73.WsWorkunitsServiceSoap version7WorkunitsServiceSoap = null;
+
+    private org.hpccsystems.ws.client.gen.wsworkunits.v1_56.WsWorkunitsServiceSoapProxy version5WorkunitsServiceSoapProxy = null;
+    private org.hpccsystems.ws.client.gen.wsworkunits.v1_69.WsWorkunitsServiceSoapProxy version6WorkunitsServiceSoapProxy = null;
+    private org.hpccsystems.ws.client.gen.wsworkunits.v1_73.WsWorkunitsServiceSoapProxy version7WorkunitsServiceSoapProxy = null;
+
+    private Version platformVersion;
+    
+    public WsWorkunitsClientSoapProxyWrapper(String baseURL, String user, String pass, Version platformVersion) {
+        this.platformVersion=platformVersion;
+        version7WorkunitsServiceSoapProxy = new WsWorkunitsServiceSoapProxy(baseURL);
+        if (version7WorkunitsServiceSoapProxy != null)
+        {
+            version7WorkunitsServiceSoap = version7WorkunitsServiceSoapProxy.getWsWorkunitsServiceSoap();
+            if (user != null && pass != null) Connection.initStub((Stub) version7WorkunitsServiceSoap, user, pass);
+        }
+
+        version5WorkunitsServiceSoapProxy = new org.hpccsystems.ws.client.gen.wsworkunits.v1_56.WsWorkunitsServiceSoapProxy(
+                baseURL);
+        if (version5WorkunitsServiceSoapProxy != null)
+        {
+            version5WorkunitsServiceSoap = version5WorkunitsServiceSoapProxy.getWsWorkunitsServiceSoap();
+            if (user != null && pass != null) Connection.initStub((Stub) version5WorkunitsServiceSoap, user, pass);
+        }
+        
+        version6WorkunitsServiceSoapProxy = new org.hpccsystems.ws.client.gen.wsworkunits.v1_69.WsWorkunitsServiceSoapProxy(
+                baseURL);
+        if (version6WorkunitsServiceSoapProxy != null)
+        {
+            version6WorkunitsServiceSoap = version6WorkunitsServiceSoapProxy.getWsWorkunitsServiceSoap();
+            if (user != null && pass != null) Connection.initStub((Stub) version6WorkunitsServiceSoap, user, pass);
+        }
+    }
+    
+    public WUCreateResponseWrapper WUCreate(WUCreateRequestWrapper params) throws Exception {
+        if (platformVersion.major==6) {

```

Comment:
Although there does seem to be a direct mapping between the WsWorkunits WSDL version to the Platform build version, as far as I know it's completely accidental...

Here's the actual mapping between hpcc build version and wsworkunits wsdl version:
5.0.0 -> 1.51
5.2.0 -> 1.54
5.4.0 -> 1.56
5.6.0 -> 1.56
6.0.0 -> 1.58
6.2.0 -> 1.62
6.4.0 -> 1.69
7.0.0 -> 1.73
7.2.0 -> 1.73
To be 100% accurate, we should determine the soapproxy version based on a plat version.major and minor range.
Also, I'll request adding a get clientversion method to all esp services in the 7.2 timeframe

Reply:
this still applies

Reply:
Currently--from what I've seen--all th wsworkunits wsdl versions that share the same major version (1.5, 1.6, 1.7) don't have any axis-breaking changes between versions 1.56, 1.58, etc.  That's why I created a single wrapper for 1.6.x that loads 1.69...it also works for 1.62.

Are you saying that you want to continue to map platform versions 6.0.0, 6.2.0 and 6.4.0 to wsworkunits wrapper version 1_6, but just want me to add additional minor version checking? Or are you saying I should create a wrapper version for every specific wsdl version (1.62 and 1.69 having separate wrappers?) I don't think the latter's necessary based on my testing. At least not for the current versions.